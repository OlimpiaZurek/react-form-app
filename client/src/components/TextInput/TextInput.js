import React  from 'react';
import PropTypes from 'prop-types';
import styles from './TextInput.module.scss';

const TextInput = ({
  onClick,
  handleChange, 
  placeholder, 
  value,
  type,
  name,
  className,
  error,
}) => (
  <div onClick={onClick} className={styles.Wrapper}>
    <input 
      type={type}
      value={value} 
      onChange={handleChange} 
      placeholder={placeholder}
      name={name}
      className={styles.Input}
      required
    />
    <div className={styles.Error}>{error}</div>
  </div>
);

TextInput.propTypes = {
  onClick: PropTypes.func,
  handleChange: PropTypes.func,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  className: PropTypes.string,
  error: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool
  ]),
};

export default TextInput;
