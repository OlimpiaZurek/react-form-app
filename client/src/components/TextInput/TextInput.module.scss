@import '../../styles/global.scss';

.Input {
    margin: $metrics-xs 0;
    padding: $metrics-s;
    width: 100%;
}

.Error {
    color: $danger-color;
}