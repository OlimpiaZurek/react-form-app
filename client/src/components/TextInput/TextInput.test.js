import React from 'react';
import { shallow, mount } from 'enzyme';
import TextInput from './TextInput.js';

const props = {
  onClick: jest.fn(),
  handleChange:  jest.fn(),
  placeholder: 'test',
  value: 'test value',
  type: 'test type',
  name: 'test name',
  className: 'test',
  error: 'test',
}

describe('<TextInput/ >', () => {
  it('should be defined', () => {
    expect(TextInput).toBeDefined();
  });

  it('should render corectly', () => {
    const wrapper = shallow(
      <TextInput {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render text input correctly with empty string value', () => {
    const wrapper = mount(
      <TextInput  {...props}  value = ''/>
    );
    expect((wrapper).prop('value')).toEqual('');
  });

  it('render text input correctly with value', () => {
    const wrapper = mount(
      <TextInput  {...props} />
    );
    expect(wrapper.props().value).toBe('test value')
  });
});
