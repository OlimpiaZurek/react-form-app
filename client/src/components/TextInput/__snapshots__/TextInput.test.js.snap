// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/ > should render corectly 1`] = `ShallowWrapper {}`;
